app/components/account-validation-form/component.js

```diff
@@ -0,0 +1,76 @@
+import Ember from 'ember';
+import service from 'ember-service/inject';
+import get from 'ember-metal/get';
+import set from 'ember-metal/set';
+import Changeset from 'ember-changeset';
+import EmailValidations from '../../validations/email';
+import lookupValidator from 'ember-changeset-validations';
+import ENV from '../../config/environment';
+import fetch from 'fetch';
+import { rejectUnsuccessfulResponses } from '../../utils/fetch-utils';
+
+const FLASH_MESSAGES = {
+  validated: 'Your email has been verified and your online account is now active.'
+};
+
+export default Ember.Component.extend({
+  flashMessages: service(),
+  accountValidated: false,
+  allowedKeys: ['email'],
+  init() {
+    this._super(...arguments);
+    set(this, 'fields', {
+      email: ''
+    });
+    set(this, 'changeset', new Changeset(get(this, 'fields'), lookupValidator(EmailValidations), EmailValidations));
+    get(this, 'changeset').validate();
+  },
+  didReceiveAttrs() {
+    this.validateNewAccount(get(this, 'username'), get(this, 'confirmation'))
+    .then(() => {
+      set(this, 'accountValidated', true);
+      this.showFlash('validated');
+    })
+    .catch((e) => {
```

Comment:
I believe this would catch by errors thrown by `validateNewAccount` or anything in the `then` above it. either way, I think you're guaranteed to get an argument.